src/zenml/cli/artifact.py

```diff
@@ -83,37 +90,76 @@ def list_artifacts(**kwargs: Any) -> None:
     help="Don't ask for confirmation.",
 )
 def delete_artifact(
-    artifact_id: str,
+    artifact_name_or_id: str,
+    version: Optional[str] = None,
     only_artifact: bool = False,
     only_metadata: bool = False,
     yes: bool = False,
 ) -> None:
-    """Delete an artifact.
+    """Delete an artifact by ID or name.
+
+    If an artifact name without a version is provided, all artifacts with that
+    name will be deleted.
 
     Args:
-        artifact_id: ID of the artifact to delete.
+        artifact_name_or_id: Name or ID of the artifact to delete.
+        version: Version of the artifact to delete.
         only_artifact: If set, only delete the artifact but not its metadata.
         only_metadata: If set, only delete metadata and not the actual artifact.
         yes: If set, don't ask for confirmation.
     """
+    is_uuid = is_valid_uuid(artifact_name_or_id)
+
     if not yes:
-        confirmation = cli_utils.confirmation(
-            f"Are you sure you want to delete artifact '{artifact_id}'?"
-        )
+        if is_uuid:
+            confirmation = cli_utils.confirmation(
+                f"Are you sure you want to delete artifact "
+                f"'{artifact_name_or_id}'?"
+            )
+        elif version:
+            confirmation = cli_utils.confirmation(
+                f"Are you sure you want to delete version '{version}' of "
+                f"artifact '{artifact_name_or_id}'?"
+            )
+        else:
+            confirmation = cli_utils.confirmation(
+                f"Are you sure you want to delete all artifacts with name "
```

Comment:
```suggestion
                f"Are you sure you want to delete all artifact versions with name "
```

And wherever else this repeats?

Reply:
I'm not getting what is the difference between this and uuid case. Aren't they the same?

Reply:
See comment below - UUID is unique for a version but name applies to multiple versions